src/azure-cli/azure/cli/command_modules/profile/_subscription_selector.py

```diff
@@ -0,0 +1,99 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from azure.cli.core._profile import (_SUBSCRIPTION_NAME, _SUBSCRIPTION_ID, _TENANT_DISPLAY_NAME, _TENANT_ID,
+                                     _IS_DEFAULT_SUBSCRIPTION)
+from knack.log import get_logger
+
+logger = get_logger(__name__)
+
+
+class SubscriptionSelector:  # pylint: disable=too-few-public-methods
+    DEFAULT_ROW_MARKER = '*'
+
+    def __init__(self, subscriptions):
+        self._subscriptions = subscriptions
+        self._active_one = None
+        self._format_subscription_table()
+
+    def _format_subscription_table(self):
+        from azure.cli.core.style import format_styled_text, Style
+        index_to_subscription_map = {}
+        table_data = []
+        subscription_name_length_limit = 36
+
+        # Sort by subscription name
+        subscriptions_sorted = sorted(self._subscriptions, key=lambda s: s[_SUBSCRIPTION_NAME].lower())
+
+        def highlight_text(text, row_is_default):
+            return format_styled_text((Style.HIGHLIGHT, text)) if row_is_default else text
+
+        for index, sub in enumerate(subscriptions_sorted, start=1):
+            # There is no need to use int, as int requires parsing. str match is sufficient.
+            index_str = str(index)  # '1', '2', ...
+            index_to_subscription_map[index_str] = sub
+
+            is_default = sub[_IS_DEFAULT_SUBSCRIPTION]
+            if is_default:
+                self._active_one = sub
+
+            # Trim subscription name if it is too long
+            subscription_name = sub[_SUBSCRIPTION_NAME]
+            if len(subscription_name) > subscription_name_length_limit:
+                subscription_name = subscription_name[:subscription_name_length_limit - 3] + '...'
+
+            row = {
+                'No': f'[{index_str}]' + (' ' + self.DEFAULT_ROW_MARKER if is_default else ''),
```

Comment:
Why `No` is not highlighted?

Reply:
UX's design.

Reply:
`No` is also highlighted in the latest code.